### PR TITLE
build: use Makefile to build Polaris instead of `build.sh`

### LINKED ISSUE #539: Polaris builds switched from build.sh to use Makefile
**What is the feature you want to add?**

**Why do you want to add this feature?**

**How to implement this feature?**

**Additional context**

### PR DESCRIPTION
**Pull request by mangoGoForward**
Signed-off-by: mango <xu.weiKyrie@foxmail.com>

**Please provide issue(s) of this PR:**
Fixes #539

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration
- [x] Docs
- [x] Installation
- [ ] Performance and Scalability
- [ ] Naming
- [ ] HealthCheck
- [ ] Test and Release

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any user-facing changes. This may include API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by mangoGoForward** [comment]
Hi @chuntaojun PTAL when you have free time

**Comment by chuntaojun** [comment]
all the logic in build.sh be moved to Makefile?

**Comment by mangoGoForward** [comment]
> all the logic in build.sh be moved to Makefile?

If I missed something, please tell me, and I will be add it.

**Comment by chuntaojun** [comment]
> > all the logic in build.sh be moved to Makefile?
> 
> If I missed something, please tell me, and I will be add it.

- build.sh 里面的逻辑应当全部转移到 makefile 里面
- build.sh 里面最终的内容是 make build 这样子的逻辑

**Comment by mangoGoForward** [comment]
> > > all the logic in build.sh be moved to Makefile?
> > 
> > 
> > If I missed something, please tell me, and I will be add it.
> 
> * build.sh 里面的逻辑应当全部转移到 makefile 里面
> * build.sh 里面最终的内容是 make build 这样子的逻辑

I think Makefile support build tool and the real logic can move into a shell file is more better, user can execute the command of `make build` to build binary and tarball, if all logic append into makefile, the readability of the code in Makefile will be worse

**Comment by chuntaojun** [comment]
> > > > all the logic in build.sh be moved to Makefile?
> > > 
> > > 
> > > If I missed something, please tell me, and I will be add it.
> > 
> > 
> > 
> > * build.sh 里面的逻辑应当全部转移到 makefile 里面
> > * build.sh 里面最终的内容是 make build 这样子的逻辑
> 
> I think Makefile support build tool and the real logic can move into a shell file is more better, user can execute the command of `make build` to build binary and tarball, like the below:
> 
> ```
> $ make help               
> 
> Usage:
>   make <target>
> 
> General
>   help             Display this help.
> 
> Build
>   build            Build binary.
> 
> $ make build VERSION=1.0.0
> bash ./build.sh 1.0.0
> GOOS is windows, binary name is polaris-server.exe
>   adding: polaris-server-release_1.0.0.windows.amd64/ (stored 0%)
>   adding: polaris-server-release_1.0.0.windows.amd64/polaris-server.yaml (deflated 72%)
>   adding: polaris-server-release_1.0.0.windows.amd64/polaris-server.exe (deflated 60%)
>   adding: polaris-server-release_1.0.0.windows.amd64/tool/ (stored 0%)
>   adding: polaris-server-release_1.0.0.windows.amd64/tool/p.bat (deflated 40%)
>   adding: polaris-server-release_1.0.0.windows.amd64/tool/start.bat (deflated 53%)
>   adding: polaris-server-release_1.0.0.windows.amd64/tool/include (deflated 69%)
>   adding: polaris-server-release_1.0.0.windows.amd64/tool/p.sh (deflated 51%)
>   adding: polaris-server-release_1.0.0.windows.amd64/tool/stop.bat (deflated 41%)
>   adding: polaris-server-release_1.0.0.windows.amd64/tool/reload.sh (deflated 52%)
>   adding: polaris-server-release_1.0.0.windows.amd64/tool/check.sh (deflated 50%)
>   adding: polaris-server-release_1.0.0.windows.amd64/tool/stop.sh (deflated 51%)
>   adding: polaris-server-release_1.0.0.windows.amd64/tool/start.sh (deflated 50%)
> ```
> 
> if all logic append into makefile, the readability of the code in Makefile will be worse

But this loses what the issue should have expected. The build thing is left to Makefile to do, and the shell only does simple things

**Comment by mangoGoForward** [comment]
> But this loses what the issue should have expected. The build thing is left to Makefile to do, and the shell only does simple things

Yes, I misunderstand the issue's expected, and I do it and followed many best practices in Makefile, like:https://github.com/kubesphere/kubesphere/blob/1f2f6157d7abfe1f0d3103cf6b88e077b3387497/Makefile#L57-L63

image

And the shell file https://github.com/kubesphere/kubesphere/blob/master/hack/gobuild.sh

**Comment by chuntaojun** [comment]
看下能不能把构建 docker 镜像的也移到 makefile 里面，

**Comment by mangoGoForward** [comment]
> 看下能不能把构建 docker 镜像的也移到 makefile 里面，

OK

**Comment by mangoGoForward** [comment]
> 看下能不能把构建 docker 镜像的也移到 makefile 里面，

Done.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# To re-generate a bundle for another specific version without changing the standard setup, you can:
+# - use the VERSION as arg of the build target (e.g make build VERSION=0.0.2)
+# - use environment variables to overwrite this value (e.g export VERSION=0.0.2)
+VERSION ?= $(shell cat version 2>/dev/null)
+
+# IMAGE_TAG defines the image tag for the docker build.
+# To re-generate a bundle for another specific version without changing the standard setup, you can:
+# - use the IMAGE_TAG as arg of the build-docker target (e.g make build-docker IMAGE_TAG=v0.0.2)
+# - use environment variables to overwrite this value (e.g export IMAGE_TAG=v0.0.2)
+IMAGE_TAG ?= v0.0.1
```
Comment (chuntaojun) [inline Makefile:28]: IMAGE_TAG = VERSION

Comment (mangoGoForward) [reply Makefile]: Done.

**Review by chuntaojun — APPROVED** [review]

**Pull request merged**